temporalio/worker/_workflow_instance.py

```diff
@@ -429,6 +448,7 @@ def activate(
             )
             # Set completion failure
             self._current_completion.failed.failure.SetInParent()
+            # TODO: Review - odd that we don't un-set success here?
```

Comment:
Wanted to get an eye on this too. Seems strange we never un-set the set-by-default success variant if there was a failure. I'm kinda surprised this doesn't cause problems, and maybe it just works by a happy accident of field ordering. Maybe it's unset somewhere I missed?

Reply:
In Python protobuf, a oneof is automatically unset when you set another of the oneof, ref https://protobuf.dev/reference/python/python-generated/#oneof